code/datums/ai/dog/dog_controller.dm

```diff
@@ -0,0 +1,167 @@
+
+
+/datum/ai_controller/dog
+	blackboard = list(BB_DOG_FETCHING = FALSE,\
+	BB_DOG_CARRY_ITEM = null,\
+	BB_DOG_THROW_LISTENERS = list(),\
+	BB_DOG_FETCH_TARGET_IGNORE = list(),\
+	BB_DOG_FETCH_TARGET = null,\
+	BB_DOG_FETCH_THROWER = null,\
+	BB_DOG_DELIVERING = FALSE,\
+	BB_DOG_FRIENDS = list())
+
+
+/datum/ai_controller/dog/TryPossessPawn(atom/new_pawn)
+	if(!isliving(new_pawn))
+		return AI_CONTROLLER_INCOMPATIBLE
+
+	RegisterSignal(new_pawn, COMSIG_ATOM_ATTACK_HAND, .proc/on_attack_hand)
+	RegisterSignal(new_pawn, COMSIG_PARENT_EXAMINE, .proc/on_examined)
+	return ..() //Run parent at end
+
+/datum/ai_controller/dog/UnpossessPawn(destroy)
+	UnregisterSignal(pawn, list(COMSIG_ATOM_ATTACK_HAND, COMSIG_PARENT_EXAMINE))
+	return ..() //Run parent at end
+
+/datum/ai_controller/dog/able_to_run()
+	var/mob/living/living_pawn = pawn
+
+	if(IS_DEAD_OR_INCAP(living_pawn))
+		return FALSE
+	return ..()
+
+/datum/ai_controller/dog/SelectBehaviors(delta_time)
+	current_behaviors = list()
+	var/mob/living/living_pawn = pawn
+
+	if(blackboard[BB_DOG_CARRY_ITEM] && blackboard[BB_DOG_FETCH_THROWER])
+		var/atom/return_target = blackboard[BB_DOG_FETCH_THROWER]
+		if(!(return_target in view(pawn, AI_DOG_THROW_LISTEN_RANGE)))
+			// if the return target isn't in sight, we'll just forget about it and carry the thing around
+			blackboard[BB_DOG_FETCH_THROWER] = null
+			return
+		current_movement_target = return_target
+		current_behaviors += GET_AI_BEHAVIOR(/datum/ai_behavior/dog_deliver)
+		blackboard[BB_DOG_DELIVERING] = TRUE
+		return
+
+	var/list/old_throw_listeners = blackboard[BB_DOG_THROW_LISTENERS]
+	var/list/new_throw_listeners = list()
+
+	// check around us for people who we can hear throw things for fetching
+	// (would probably be smart to make this run less frequently than every tick)
```

Comment:
theres a var for this now

Reply:
also why dont you just way for someone to say "Fetch" and then listen for that person to throw something?

Reply:
1. Not intuitive
2. Less people would unknowingly throw grenades for Ian to bring them back.